### PR TITLE
Add `IDocumentProvider` service

### PR DESCRIPTION
**Pull request by dougbu**
- aspnet/AspNetCore#4948

/cc @domaindrivendev 
/fyi @mkartak

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.SwaggerGen/Generator/IDocumentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ApiDescription
+{
+    /// <summary>
+    /// his service will be looked up by name from the service collection when using
```
Comment (mkArtakMSFT) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Generator/IDocumentProvider.cs]: Type: "This ..."

File: src/Swashbuckle.AspNetCore.SwaggerGen/Generator/DocumentProvider.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            _options = options.Value;
+            _swaggerProvider = swaggerProvider;
+        }
+
+        public Task GenerateAsync(string documentName, TextWriter writer)
+        {
+            // Let UnknownSwaggerDocument or other exception bubble up to caller.
+            var swagger = _swaggerProvider.GetSwagger(documentName, host: null, basePath: null);
+            var jsonWriter = new OpenApiJsonWriter(writer);
```
Comment (mkArtakMSFT) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Generator/DocumentProvider.cs]: Should this be disposed?

File: src/Swashbuckle.AspNetCore.SwaggerGen/Application/SwaggerGenOptionsExtensions.cs [hunk 3]
```diff
@@ -1,9 +1,9 @@
 using System;
 using System.Collections.Generic;
 using System.Xml.XPath;
-using Microsoft.AspNetCore.Mvc.ApiExplorer;
 using Microsoft.OpenApi.Models;
 using Swashbuckle.AspNetCore.SwaggerGen;
+using ApiDescriptionType = Microsoft.AspNetCore.Mvc.ApiExplorer.ApiDescription;
```
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Application/SwaggerGenOptionsExtensions.cs]: What's the reason for aliasing `ApiDescription` here - is there a conflict? Would be good to understand the need for this.

Comment (dougbu) [reply src/Swashbuckle.AspNetCore.SwaggerGen/Application/SwaggerGenOptionsExtensions.cs]: Yes, we introduced a conflict when we created the `IDocumentProvider` interface -- that namespace conflicts with this type name.

File: src/Swashbuckle.AspNetCore.SwaggerGen/Generator/DocumentProvider.cs (hunk 2 above)
Comment (dougbu) [reply src/Swashbuckle.AspNetCore.SwaggerGen/Generator/DocumentProvider.cs]: No, the `OpenApiJsonWriter` is not disposable

**Comment by dougbu** [comment]
:up::date: to rebase and fix the typo @mkArtakMSFT 

@domaindrivendev over to you

**Pull request merged**